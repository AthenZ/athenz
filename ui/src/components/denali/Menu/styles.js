/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { css } from 'emotion';
import { cssFontStyles } from '../styles/fonts';
import { colors } from '../styles/colors';
import { cssDropShadow } from '../styles/drop-shadow';

export const makeCssPopperBox = (props) => css`
    ${cssDropShadow};
    ${cssFontStyles.default};
    background-color: ${colors.white};
    border: 2px solid ${colors.border};
    border-radius: 4px;
    color: ${colors.black};
    cursor: default;
    line-height: 1.4;
    margin: 0;
    max-width: ${props.basic ? '400px' : undefined};
    min-width: 20px;
    padding: ${props.basic ? '6px 10px' : 0};
    transition: ${!props.noanim && 'opacity 0.25s ease-in'};
    z-index: 999;
    label: denali-menu;
    /* Shift the popper to allow space for arrow */
    &[data-placement*='bottom'] {
        margin-top: 8px;
    }
    &[data-placement*='top'] {
        margin-bottom: 8px;
    }
    &[data-placement*='left'] {
        margin-right: 8px;
    }
    &[data-placement*='right'] {
        margin-left: 8px;
    }

    /**
   * Invisible padding between trigger and main content (around the arrow) so
   * user can slide from one to the other without the menu closing.
   */
    &::after {
        background: transparent;
        content: '';
        position: absolute;
    }
    &[data-placement*='top']::after {
        bottom: -12px;
        height: 12px;
        width: 100%;
    }
    &[data-placement*='bottom']::after {
        height: 12px;
        top: -12px;
        width: 100%;
    }
    &[data-placement*='left']::after {
        right: -12px;
        height: 100%;
        width: 12px;
    }
    &[data-placement*='right']::after {
        left: -12px;
        height: 100%;
        width: 12px;
    }
`;

export const cssArrow = css`
    height: 1rem;
    position: absolute;
    width: 1rem;
    label: menu-arrow;

    &::before,
    &::after {
        border-style: solid;
        content: '';
        display: block;
        height: 0;
        margin: auto;
        position: absolute;
        width: 0;
    }

    &[data-placement*='bottom'] {
        left: 0;
        margin-top: -6px;
        top: -8px;
        width: 1rem;
    }

    &[data-placement*='bottom']::before {
        border-color: transparent transparent ${colors.grey400} transparent;
        border-width: 0 8px 8px 8px;
        position: absolute;
        top: 4px;
    }

    &[data-placement*='bottom']::after {
        border-color: transparent transparent ${colors.white} transparent;
        border-width: 0 8px 8px 8px;
        top: 7px;
    }

    &[data-placement*='top'] {
        bottom: 0;
        height: 1rem;
        left: 0;
        margin-bottom: -1rem;
        width: 1rem;
    }

    &[data-placement*='top']::before {
        border-color: ${colors.grey400} transparent transparent transparent;
        border-width: 8px 8px 0 8px;
        position: absolute;
        top: 2px;
    }

    &[data-placement*='top']::after {
        border-color: ${colors.white} transparent transparent transparent;
        border-width: 8px 8px 0 8px;
        top: -1px;
    }

    &[data-placement*='right'] {
        height: 1rem;
        left: 0;
        margin-left: -1rem;
        width: 1rem;
    }

    &[data-placement*='right']::before {
        border-color: transparent ${colors.grey400} transparent transparent;
        border-width: 8px 8px 8px 0;
        left: 5px;
    }

    &[data-placement*='right']::after {
        border-color: transparent ${colors.white} transparent transparent;
        border-width: 8px 8px 8px 0;
        left: 8px;
        top: 0;
    }

    &[data-placement*='left'] {
        height: 1rem;
        margin-right: -1rem;
        right: 0;
        width: 1rem;
    }

    &[data-placement*='left']::before {
        border-color: transparent transparent transparent ${colors.grey400};
        border-width: 8px 0 8px 8px;
        left: 2px;
    }

    &[data-placement*='left']::after {
        border-color: transparent transparent transparent ${colors.white};
        border-width: 8px 0 8px 8px;
        left: -1px;
        top: 0;
    }
`;
