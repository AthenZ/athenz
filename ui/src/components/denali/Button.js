/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { cx, css } from 'emotion';
import { rgba } from 'polished';
import { colors } from './styles/colors';
import { cssFontFamilies, cssFontWeights } from './styles/fonts';

// NOTE: Psuedo class priority:
// Link Visited Hover Focus Active
// "Lord Vader Hates Furry Animals"

const buttonBase = (props) => css`
    border: none;
    border-radius: 2px;
    box-shadow: 0 0 0 1px transparent inset, 0 0 0 0 ${colors.grey500} inset;
    box-sizing: border-box;
    cursor: pointer;
    display: inline-block;
    ${cssFontFamilies.default};
    ${cssFontWeights.normal};
    font-size: ${props.size === 'large'
        ? '18px'
        : props.size === 'regular'
        ? '14px'
        : '12px'};
    line-height: 1;
    margin: 5px;
    min-width: ${props.size === 'small' ? '70px' : undefined};
    outline: 0;
    padding: ${props.size === 'large'
        ? '12px 28px'
        : props.size === 'regular'
        ? '10px 24px'
        : '8px 18px'};
    text-align: center;
    text-shadow: none;
    text-transform: none;
    transition: ${!props.noanim ? 'all 0.2s ease-in' : undefined};
    vertical-align: baseline;
    white-space: nowrap;
    label: button;
    &:first-of-type {
        margin-left: 0;
    }
    &:disabled {
        background: ${props.dark
            ? rgba(colors.white, 0.1)
            : rgba(colors.grey800, 0.1)};
        color: ${props.dark
            ? rgba(colors.white, 0.2)
            : rgba(colors.grey800, 0.2)};
        cursor: not-allowed;
    }
`;

const buttonVariants = {
    primary: (props) => css`
        background: linear-gradient(
            to right,
            ${props.active ? colors.brand700 : colors.brand600},
            ${props.active ? colors.brand800 : colors.brand700}
        );
        color: ${colors.white};
        &:hover:not(:disabled) {
            background: linear-gradient(
                to right,
                ${colors.brand700},
                ${colors.brand800}
            );
        }
    `,
    secondary: (props) => css`
        background: ${props.dark
            ? props.active
                ? `linear-gradient(to right, ${rgba(
                      colors.brand700,
                      0.3
                  )}, ${rgba(colors.brand800, 0.3)})`
                : 'transparent'
            : props.active
            ? colors.brand200
            : 'transparent'};
        box-shadow: 0 0 0 1px ${colors.brand600} inset;
        color: ${props.dark ? colors.white : colors.brand600};
        &:disabled {
            box-shadow: ${props.dark
                ? `0 0 0 1px ${rgba(colors.white, 0.2)} inset`
                : `0 0 0 1px ${rgba(colors.grey800, 0.1)} inset`};
        }
        &:hover:not(:disabled) {
            background: ${props.dark
                ? `linear-gradient(to right, ${rgba(
                      colors.brand700,
                      0.3
                  )}, ${rgba(colors.brand800, 0.3)})`
                : colors.brand200};
        }
    `,
    danger: () => css`
        background: ${colors.red500};
        color: ${colors.white};
        &:hover:not(:disabled) {
            background: ${colors.red600};
        }
    `,
};

class Button extends React.PureComponent {
    render() {
        const {
            active,
            className,
            danger,
            dark,
            noanim,
            secondary,
            size,
            ...rest
        } = this.props;

        const cssProps = { active, dark, noanim, size };

        const makeCssButton = (props) => css`
            ${buttonBase(props)};
            ${(danger
                ? buttonVariants.danger
                : secondary
                ? buttonVariants.secondary
                : buttonVariants.primary)(props)};
        `;
        const cssButton = makeCssButton(cssProps);

        const classes = cx(cssButton, 'denali-button', className);

        return <button className={classes} {...rest} />;
    }
}

Button.propTypes = {
    /** Set active state */
    active: PropTypes.bool,
    /** Additonal class to apply to the button */
    className: PropTypes.string,
    /** Error color scheme */
    danger: PropTypes.bool,
    /** Dark color scheme (can be used in conjunction with 'secondary') */
    dark: PropTypes.bool,
    /** Disable the button (no clicks) */
    disabled: PropTypes.bool,
    /** Forward React ref to the `<input>` element */
    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
    /** Name of the `button` element */
    name: PropTypes.string,
    /** Disable animations / transitions */
    noanim: PropTypes.bool,
    /** Secondary color scheme */
    secondary: PropTypes.bool,
    /** Size of buttons (default is regular) */
    size: PropTypes.oneOf(['large', 'regular', 'small']),
    /** <button> type attribute */
    type: PropTypes.oneOf(['button', 'submit', 'reset']),
    /** <button> value attribute */
    value: PropTypes.any,
    /** Gets called when the button is clicked */
    onClick: PropTypes.func,
};

Button.defaultProps = {
    active: false,
    danger: false,
    dark: false,
    disabled: false,
    noanim: false,
    size: 'regular',
    secondary: false,
};

export default Button;
