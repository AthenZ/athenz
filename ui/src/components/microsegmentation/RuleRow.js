/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import Icon from '../denali/icons/Icon';
import { colors } from '../denali/styles';
import styled from '@emotion/styled';
import DeleteModal from '../modal/DeleteModal';
import Menu from '../denali/Menu/Menu';
import DateUtils from '../utils/DateUtils';
import RequestUtils from '../utils/RequestUtils';
import ServiceList from './ServiceList';
import { css, keyframes } from '@emotion/react';
import EnforcementStateList from './EnforcementStateList';
import AddSegmentation from './AddSegmentation';

const TDStyled = styled.td`
    background-color: ${(props) => props.color};
    text-align: ${(props) => props.align};
    padding: 5px 0 5px 15px;
    vertical-align: middle;
    word-break: break-all;
`;

const GroupTDStyled = styled.td`
    background-color: ${(props) => props.color};
    text-align: ${(props) => props.align};
    padding: 5px 0 5px 15px;
    vertical-align: middle;
    word-break: break-all;
    text-decoration: dashed underline;
`;

const colorTransition = keyframes`
        0% {
            background-color: rgba(21, 192, 70, 0.20);
        }
        100% {
            background-color: transparent;
        }
`;

const TrStyled = styled.tr`
    ${(props) =>
        props.isSuccess &&
        css`
            animation: ${colorTransition} 3s ease;
        `}
`;

const MenuDiv = styled.div`
    padding: 5px 10px;
    background-color: black;
    color: white;
    font-size: 12px;
`;

export default class RuleRow extends React.Component {
    constructor(props) {
        super(props);
        this.api = this.props.api;
        this.onSubmitDelete = this.onSubmitDelete.bind(this);
        this.onClickDeleteCancel = this.onClickDeleteCancel.bind(this);
        this.onClickDeleteCondition = this.onClickDeleteCondition.bind(this);
        this.onClickEditCancel = this.onClickEditCancel.bind(this);
        this.onClickEditSubmit = this.onClickEditSubmit.bind(this);
        this.onClickEdit = this.onClickEdit.bind(this);
        this.saveJustification = this.saveJustification.bind(this);
        this.state = {
            deleteName:
                this.props.category === 'inbound'
                    ? this.props.details['destination_service']
                    : this.props.details['source_service'],
            showDelete: false,
            assertionId: this.props.details['assertionIdx'],
            port: '',
            showDeleteAssertionCondition: false,
            policyName: '',
            conditionId: '',
            justification: '',
            showEditSegmentation: false,
        };
        this.localDate = new DateUtils();
    }

    saveJustification(val) {
        this.setState({ justification: val });
    }

    onClickDelete(name, id, port) {
        this.setState({
            showDelete: true,
            deleteName: name,
            assertionId: id,
            port: port,
        });
    }

    onClickDeleteCondition(assertionId, conditionId, policyName) {
        this.setState({
            showDeleteAssertionCondition: true,
            policyName: policyName,
            assertionId: assertionId,
            conditionId: conditionId,
        });
    }

    onClickEdit() {
        this.setState({
            showEditSegmentation: true,
        });
    }

    onClickEditCancel() {
        this.setState({
            showEditSegmentation: false,
        });
    }

    onClickEditSubmit() {
        this.setState({
            showEditSegmentation: false,
        });
        this.props.onUpdateSuccess();
    }

    onSubmitDelete(domain) {
        if (
            this.props.justificationRequired &&
            (this.state.justification === undefined ||
                this.state.justification.trim() === '')
        ) {
            this.setState({
                errorMessage: 'Justification is required to delete ACL Policy.',
            });
            return;
        }

        let deletePolicyName =
            'acl.' + this.state.deleteName + '.' + this.props.category;
        let deleteRoleName =
            deletePolicyName + '-' + this.props.details['identifier'];
        Promise.all([
            this.api.deleteAssertion(
                domain,
                deletePolicyName,
                this.state.assertionId,
                this.state.justification
                    ? this.state.justification
                    : 'Micro-segmentaion assertion deletion',
                this.props._csrf
            ),
            this.api.deleteRole(
                domain,
                deleteRoleName,
                this.state.justification
                    ? this.state.justification
                    : 'Micro-segmentaion Role deletion',
                this.props._csrf
            ),
        ])
            .then(() => {
                this.setState({
                    showDelete: false,
                });
                this.props.onUpdateSuccess();
            })
            .catch((err) => {
                if (err.statusCode === 404) {
                    this.setState({
                        showDelete: false,
                    });
                    this.props.onUpdateSuccess();
                } else {
                    this.setState({
                        errorMessage: RequestUtils.xhrErrorCheckHelper(err),
                    });
                }
            });
    }

    onClickDeleteCancel() {
        this.setState({
            showDelete: false,
            deleteName: '',
            errorMessage: null,
        });
    }

    onSubmitDeleteCondition() {
        if (
            this.props.justificationRequired &&
            (this.state.justification === undefined ||
                this.state.justification.trim() === '')
        ) {
            this.setState({
                errorMessage:
                    'Justification is required to delete Policy Enforcement Condition.',
            });
            return;
        }

        this.api
            .deleteAssertionCondition(
                this.props.domain,
                this.state.policyName,
                this.state.assertionId,
                this.state.conditionId,
                this.state.justification
                    ? this.state.justification
                    : 'Micro-segmentaion Assertion Condition deletion',
                this.props._csrf
            )
            .then(() => {
                this.setState({
                    showDeleteAssertionCondition: false,
                    errorMessage: null,
                });
                this.props.onUpdateSuccess();
            })
            .catch((err) => {
                this.setState({
                    errorMessage: RequestUtils.xhrErrorCheckHelper(err),
                });
            });
    }

    onClickDeleteConditionCancel() {
        this.setState({
            showDeleteAssertionCondition: false,
            errorMessage: null,
        });
    }

    render() {
        let rows = [];
        let left = 'left';
        let center = 'center';
        let data = this.props.details;
        let color = this.props.color;
        let key = '';
        let submitDelete = this.onSubmitDelete.bind(this, this.props.domain);
        let submitDeleteCondition = this.onSubmitDeleteCondition.bind(this);
        let clickDeleteCancel = this.onClickDeleteCancel.bind(this);
        let clickDeleteConditionCancel =
            this.onClickDeleteConditionCancel.bind(this);
        let inbound = this.props.category === 'inbound';
        let clickDelete;
        if (inbound) {
            key =
                this.props.category +
                data.destination_service +
                data.destination_port +
                this.props.idx;
            clickDelete = this.onClickDelete.bind(
                this,
                this.state.deleteName,
                this.state.assertionId,
                data.destination_port
            );
        } else {
            key =
                this.props.category +
                data.source_service +
                data.source_port +
                this.props.idx;
            clickDelete = this.onClickDelete.bind(
                this,
                this.state.deleteName,
                this.state.assertionId,
                data.source_port
            );
        }

        let editData = this.props.details;
        editData['category'] = this.props.category;
        let addSegmentation = this.state.showEditSegmentation ? (
            <AddSegmentation
                api={this.api}
                domain={this.props.domain}
                onSubmit={this.onClickEditSubmit}
                onCancel={this.onClickEditCancel}
                _csrf={this.props._csrf}
                showAddSegment={this.state.showEditSegmentation}
                justificationRequired={this.props.justificationRequired}
                editMode={true}
                data={editData}
            />
        ) : (
            ''
        );

        rows.push(
            <TrStyled key={key} data-testid='segmentation-row'>
                <TDStyled color={color} align={left}>
                    {data['identifier']}
                </TDStyled>

                {inbound && (
                    <TDStyled color={color} align={left}>
                        {data['destination_service']}
                    </TDStyled>
                )}
                {!inbound && (
                    <TDStyled color={color} align={left}>
                        {data['source_service']}
                    </TDStyled>
                )}

                {inbound && (
                    <TDStyled color={color} align={left}>
                        {data['destination_port']}
                    </TDStyled>
                )}
                {!inbound && (
                    <TDStyled color={color} align={left}>
                        {data['source_port']}
                    </TDStyled>
                )}

                <GroupTDStyled color={color} align={left}>
                    <Menu
                        placement='right'
                        boundary='scrollParent'
                        trigger={
                            <span>
                                <Icon
                                    icon={'service-setting'}
                                    color={colors.icons}
                                    isLink
                                    size={'1.25em'}
                                    verticalAlign={'text-bottom'}
                                />
                            </span>
                        }
                    >
                        {inbound && (
                            <ServiceList
                                list={data['source_services']}
                                api={this.api}
                                domain={this.props.domain}
                            />
                        )}
                        {!inbound && (
                            <ServiceList
                                list={data['destination_services']}
                                api={this.api}
                                domain={this.props.domain}
                            />
                        )}
                    </Menu>
                </GroupTDStyled>

                {inbound && (
                    <TDStyled color={color} align={left}>
                        {data['source_port']}
                    </TDStyled>
                )}
                {!inbound && (
                    <TDStyled color={color} align={left}>
                        {data['destination_port']}
                    </TDStyled>
                )}

                <TDStyled color={color} align={left}>
                    {data['layer']}
                </TDStyled>

                <GroupTDStyled color={color} align={center}>
                    <Menu
                        placement='right'
                        boundary='scrollParent'
                        triggerOn='click'
                        trigger={
                            <span>
                                <Icon
                                    icon={
                                        this.props.category === 'inbound'
                                            ? 'network-resource'
                                            : 'network-role'
                                    }
                                    color={colors.icons}
                                    isLink
                                    size={'1.25em'}
                                    verticalAlign={'text-bottom'}
                                />
                            </span>
                        }
                    >
                        <EnforcementStateList
                            list={data['conditionsList']}
                            api={this.api}
                            domain={this.props.domain}
                            _csrf={this.props._csrf}
                            deleteCondition={this.onClickDeleteCondition}
                        />
                    </Menu>
                </GroupTDStyled>
                <TDStyled color={color} align={center}>
                    <Menu
                        placement='bottom-start'
                        trigger={
                            <span>
                                <Icon
                                    icon={'pencil'}
                                    onClick={this.onClickEdit}
                                    color={colors.icons}
                                    isLink
                                    size={'1.25em'}
                                    verticalAlign={'text-bottom'}
                                />
                            </span>
                        }
                    >
                        <MenuDiv>Edit Rule</MenuDiv>
                    </Menu>
                </TDStyled>
                {addSegmentation}
                <TDStyled color={color} align={center}>
                    <Menu
                        placement='bottom-start'
                        trigger={
                            <span>
                                <Icon
                                    icon={'trash'}
                                    onClick={clickDelete}
                                    color={colors.icons}
                                    isLink
                                    size={'1.25em'}
                                    verticalAlign={'text-bottom'}
                                />
                            </span>
                        }
                    >
                        <MenuDiv>Delete Rule</MenuDiv>
                    </Menu>
                </TDStyled>
            </TrStyled>
        );

        if (this.state.showDelete) {
            rows.push(
                <DeleteModal
                    name={this.props.details['source_service']}
                    isOpen={this.state.showDelete}
                    cancel={clickDeleteCancel}
                    submit={submitDelete}
                    key={this.props.details['source_service'] + '-delete'}
                    message={
                        'Are you sure you want to permanently delete the ' +
                        this.props.category +
                        ' rule '
                    }
                    errorMessage={this.state.errorMessage}
                    showJustification={this.props.justificationRequired}
                    onJustification={this.saveJustification}
                />
            );
        }

        if (this.state.showDeleteAssertionCondition) {
            rows.push(
                <DeleteModal
                    isOpen={this.state.showDeleteAssertionCondition}
                    cancel={clickDeleteConditionCancel}
                    submit={submitDeleteCondition}
                    key={
                        this.state.assertionId +
                        '-' +
                        this.state.conditionId +
                        '-delete'
                    }
                    message={
                        'Are you sure you want to permanently delete the Policy Enforcement Condition'
                    }
                    errorMessage={this.state.errorMessage}
                    showJustification={this.props.justificationRequired}
                    onJustification={this.saveJustification}
                />
            );
        }

        return rows;
    }
}
