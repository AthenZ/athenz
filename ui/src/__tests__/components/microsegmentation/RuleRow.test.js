/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render } from '@testing-library/react';
import { colors } from '../../../components/denali/styles';
import RuleRow from "../../../components/microsegmentation/RuleRow";

describe('RuleRow', () => {
    it('should render', () => {
        let details = {
            "source_service": "serviceA",
            "source_port": "1111",
            "destination_port": "2222",
            "destination_services": [
                "serviceB",
                "serviceC",
                "serviceD"
            ],
            "layer" : "tcp"
        };
        let domain = 'domain';
        let role = 'roleName';
        let color = colors.row;
        let idx = '50';
        const { getByTestId } = render(
            <table>
                <tbody>
                   <RuleRow
                       category={'inbound'}
                       domain={domain}
                       details={details}
                       idx={idx}
                       color={color}
                       key={idx}
                       _csrf={'_csrf'}
                   />
                </tbody>
            </table>
        );
        const ruleRow = getByTestId('segmentation-row');

        expect(ruleRow).toMatchSnapshot();
    });
});
