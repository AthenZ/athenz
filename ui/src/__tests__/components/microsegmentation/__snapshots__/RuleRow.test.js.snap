// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleRow should render 1`] = `
.emotion-2 {
  background-color: #3570f40D;
  text-align: left;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
}

.emotion-8 {
  background-color: #3570f40D;
  text-align: left;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
  -webkit-text-decoration: dashed underline;
  text-decoration: dashed underline;
}

.emotion-10 {
  fill: #188fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-15 {
  background-color: #3570f40D;
  text-align: center;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
  -webkit-text-decoration: dashed underline;
  text-decoration: dashed underline;
}

.emotion-18 {
  background-color: #3570f40D;
  text-align: center;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
}

<tr
  class="emotion-0 emotion-1"
  data-testid="segmentation-row"
>
  <td
    class="emotion-2 emotion-3"
    color="#3570f40D"
  />
  <td
    class="emotion-2 emotion-3"
    color="#3570f40D"
  />
  <td
    class="emotion-2 emotion-3"
    color="#3570f40D"
  >
    2222
  </td>
  <td
    class="emotion-8 emotion-9"
    color="#3570f40D"
  >
    <span>
      <svg
        class="emotion-10"
        data-testid="icon"
        height="1.25em"
        viewBox="0 0 1024 1024"
        width="1.25em"
      >
        <title>
          service-setting
        </title>
        <path
          d="M612.267 401.067l-38.4 61.867 29.867 29.867 61.867-38.4 21.333 8.533 17.067 70.4h42.667l17.067-70.4 21.333-8.533 61.867 38.4 29.867-29.867-38.4-61.867c3.496-6.133 6.416-13.229 8.393-20.708l0.14-0.625 70.4-17.067v-42.667l-70.4-17.067c-2.117-8.104-5.037-15.2-8.765-21.778l0.232 0.445 38.4-61.867-29.867-29.867-61.867 38.4-21.333-8.533-17.067-70.4h-42.667l-17.067 70.4-21.333 8.533-61.867-38.4-29.867 29.867 38.4 61.867c-3.496 6.133-6.416 13.229-8.393 20.708l-0.14 0.625-70.4 17.067v42.667l70.4 17.067c2.117 8.104 5.037 15.2 8.765 21.778l-0.232-0.445zM725.333 298.667c23.564 0 42.667 19.103 42.667 42.667s-19.103 42.667-42.667 42.667c-23.564 0-42.667-19.103-42.667-42.667v0c0-23.564 19.103-42.667 42.667-42.667v0z"
        />
        <path
          d="M900.267 612.267c-18.864-15.311-43.169-24.582-69.64-24.582-5.536 0-10.978 0.406-16.297 1.189l0.603-0.073-211.2 38.4c0.002-0.241 0.003-0.525 0.003-0.81 0-27.37-11.454-52.065-29.83-69.553l-0.040-0.038-151.467-151.467h-315.733c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h281.6l125.867 125.867c3.276 3.276 5.302 7.801 5.302 12.8 0 9.997-8.105 18.102-18.102 18.102-4.999 0-9.524-2.026-12.8-5.302l-74.667-74.667c-7.042-5.327-15.947-8.533-25.6-8.533-23.564 0-42.667 19.103-42.667 42.667 0 9.653 3.206 18.558 8.611 25.707l-0.078-0.107 74.667 74.667c17.528 18.44 42.239 29.911 69.629 29.911 1.021 0 2.039-0.016 3.052-0.048l-0.148 0.004c13.761-0.062 26.955-2.401 39.254-6.657l-0.854 0.257 290.133-51.2c1.148-0.219 2.469-0.345 3.82-0.345 5.025 0 9.645 1.738 13.29 4.645l-0.043-0.033c3.941 3.531 6.41 8.637 6.41 14.32 0 0.216-0.004 0.431-0.011 0.645l0.001-0.031c0.003 0.128 0.004 0.279 0.004 0.43 0 10.273-7.262 18.851-16.932 20.879l-0.139 0.024-396.8 72.533-232.533-189.867h-100.267c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h70.4l236.8 194.133 437.333-78.933c49.475-9.916 86.226-53.011 86.226-104.69 0-32.293-14.35-61.234-37.022-80.794l-0.138-0.116z"
        />
      </svg>
    </span>
  </td>
  <td
    class="emotion-2 emotion-3"
    color="#3570f40D"
  >
    1111
  </td>
  <td
    class="emotion-2 emotion-3"
    color="#3570f40D"
  >
    tcp
  </td>
  <td
    class="emotion-15 emotion-9"
    color="#3570f40D"
  >
    <span>
      <svg
        class="emotion-10"
        data-testid="icon"
        height="1.25em"
        viewBox="0 0 1024 1024"
        width="1.25em"
      >
        <title>
          network-resource
        </title>
        <path
          d="M170.667 213.333h388.267l29.867 29.867-55.467 55.467h170.667v-170.667l-55.467 55.467-55.467-55.467h-422.4c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0z"
        />
        <path
          d="M588.8 780.8l-29.867 29.867h-388.267c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h422.4l55.467-55.467 55.467 55.467v-170.667h-170.667z"
        />
        <path
          d="M170.667 554.667h256v85.333l128-128-128-128v85.333h-256c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0z"
        />
        <path
          d="M768 341.333c-94.257 0-170.667 76.41-170.667 170.667s76.41 170.667 170.667 170.667c94.257 0 170.667-76.41 170.667-170.667v0c0-94.257-76.41-170.667-170.667-170.667v0zM768 597.333c-47.128 0-85.333-38.205-85.333-85.333s38.205-85.333 85.333-85.333c47.128 0 85.333 38.205 85.333 85.333v0c0 47.128-38.205 85.333-85.333 85.333v0z"
        />
      </svg>
    </span>
  </td>
  <td
    class="emotion-18 emotion-3"
    color="#3570f40D"
  >
    <span>
      <svg
        class="emotion-10"
        data-testid="icon"
        height="1.25em"
        viewBox="0 0 1024 1024"
        width="1.25em"
      >
        <title>
          pencil
        </title>
        <path
          d="M925.867 401.067l-300.8-302.933c-7.434-8.083-18.061-13.13-29.867-13.13s-22.432 5.047-29.84 13.101l-0.026 0.029-467.2 467.2c-7.763 7.612-12.628 18.155-12.8 29.835l-0 0.032v300.8c0 23.564 19.103 42.667 42.667 42.667v0h300.8c11.712-0.172 22.255-5.037 29.86-12.793l0.007-0.007 467.2-465.067c8.083-7.434 13.13-18.061 13.13-29.867s-5.047-22.432-13.101-29.84l-0.029-0.026zM411.733 853.333h-241.067v-241.067l266.667-264.533 238.933 241.067zM738.133 529.067l-241.067-241.067 98.133-100.267 241.067 243.2z"
        />
      </svg>
    </span>
  </td>
  
  <td
    class="emotion-18 emotion-3"
    color="#3570f40D"
  >
    <span>
      <svg
        class="emotion-10"
        data-testid="icon"
        height="1.25em"
        viewBox="0 0 1024 1024"
        width="1.25em"
      >
        <title>
          trash
        </title>
        <path
          d="M917.333 187.733c0.007 0.319 0.012 0.694 0.012 1.071 0 12.755-4.867 24.374-12.846 33.1l0.034-0.038c-6.696 7.89-16.62 12.864-27.706 12.864-0.76 0-1.515-0.023-2.264-0.069l0.103 0.005h-256c-23.564 0-42.667-19.103-42.667-42.667v0-42.667h-128v42.667c0 23.564-19.103 42.667-42.667 42.667v0h-253.867c-0.136 0.001-0.298 0.002-0.459 0.002-22.485 0-41.102-16.565-44.311-38.157l-0.030-0.245c-0.007-0.319-0.012-0.694-0.012-1.071 0-12.755 4.867-24.374 12.846-33.1l-0.034 0.038c6.696-7.89 16.62-12.864 27.706-12.864 0.76 0 1.515 0.023 2.264 0.069l-0.103-0.005h213.333v-42.667c0-23.564 19.103-42.667 42.667-42.667v0h213.333c23.564 0 42.667 19.103 42.667 42.667v0 42.667h211.2c0.136-0.001 0.298-0.002 0.459-0.002 22.485 0 41.102 16.565 44.311 38.157l0.030 0.245z"
        />
        <path
          d="M742.4 358.4l-51.2 516.267h-358.4l-51.2-516.267c-2.24-21.656-20.391-38.401-42.453-38.401-0.075 0-0.15 0-0.225 0.001l0.012-0c-0.064-0-0.139-0.001-0.214-0.001-23.564 0-42.667 19.103-42.667 42.667 0 1.503 0.078 2.988 0.229 4.45l-0.015-0.183 55.467 554.667c2.24 21.656 20.391 38.401 42.453 38.401 0.075 0 0.15-0 0.225-0.001l-0.012 0h435.2c0.064 0 0.139 0.001 0.214 0.001 22.062 0 40.213-16.744 42.437-38.218l0.015-0.183 55.467-554.667c0.136-1.28 0.214-2.764 0.214-4.267 0-23.564-19.103-42.667-42.667-42.667-0.075 0-0.15 0-0.226 0.001l0.012-0c-0.064-0-0.139-0.001-0.214-0.001-22.062 0-40.213 16.744-42.437 38.218l-0.015 0.183z"
        />
      </svg>
    </span>
  </td>
</tr>
`;
